sdk/core/AzureCore/Source/DataStructures/Collections.swift

```diff
@@ -228,18 +228,11 @@ public class PagedCollection<SingleElement: Codable> {
         else { return }
 
         // Create a new cancellation token per page request from the old one, or the global default, if set
-        if let cancellationTimeout = (context.value(forKey: .cancellationToken) as? CancellationToken)?
-            .timeoutInSeconds {
-            context.add(
-                value: CancellationToken(timeoutInSeconds: cancellationTimeout) as AnyObject,
-                forKey: .cancellationToken
-            )
-        } else if let globalTimeout = client.commonOptions.transportOptions.timeoutInSeconds {
-            context.add(
-                value: CancellationToken(timeoutInSeconds: globalTimeout) as AnyObject,
-                forKey: .cancellationToken
-            )
+        let cancellationToken = CancellationToken()
```

Comment:
I'm pretty sure this doesn't work. Here we're replacing the customer's provided token with a new one, copying over the timeout. If the customer calls `cancel()` on the token they have, nothing will happen.

Reply:
Good point. Originally I had a restart method but I was concerned what would happen if I just reused the token. If I reset and restarted the timer after it had already been started, I could envision a frustrating race condition. I'll try it out and see how that works. Otherwise we'd need to return a cancellation token, which would be not great.

Reply:
I basically use an eTag concept to accomplish this. The timer captures the value at the time it is started so if reset is called, the value will change and it will not mark the token as cancelled.